### PR TITLE
First Implementation of Functions and Tests

### PR DESCRIPTION
**Pull request by minight**
Implemented insertBefore, insertAfter, and delete.
Modified the showList function to display the list in a single line.

Implemented basic tests for implemented functions. They look right enough.

### CODE REVIEW

File: week03/DLList.c [hunk 1]
```diff
@@ -107,7 +107,8 @@ void showDLList(FILE *out, DLList L)
     assert(out != NULL); assert(L != NULL);
     DLListNode *curr;
     for (curr = L->first; curr != NULL; curr = curr->next)
-        fprintf(out,"%s\n",curr->value);
+        fprintf(out,"%s - ",curr->value);
+    printf("\n");
```
Comment (kotarou3) [inline week03/DLList.c]: you wot

File: week03/DLList.c [hunk 2]
```diff
@@ -209,7 +210,24 @@ int DLListMoveTo(DLList L, int i)
 void DLListBefore(DLList L, char *it)
 {
     assert(L != NULL);
+    DLListNode *new = newDLListNode(it);
+    if(L->curr == L->first){ // case, where current is first
```
Comment (kotarou3) [inline week03/DLList.c]: spaces before and after parenthesis please (except for function calls)

File: week03/testList.c [hunk 3]
```diff
@@ -6,11 +6,75 @@
...
+	printf("===== Testing tInsertBefore =====\n");
+	printf("Inserts integers 1-10 into the List before each other\n");
+	printf("Current Element: %s\n", DLListCurrent(L));
+	printf("Expected Output: 10 9 8 7 6 5 4 3 2 1 %s", DLListCurrent(L));
+	showDLList(stdout,L);
+	for (int i = 1; i <= 10; i++){
+		char input [30];
+		sprintf(input, "%d", i);
+		DLListBefore(L, input);
+	}
```
Comment (kotarou3) [inline week03/testList.c]: Nice tabs (please change to spaces)

File: week03/DLList.c [hunk 4]
```diff
@@ -209,7 +210,24 @@ int DLListMoveTo(DLList L, int i)
...
+        L->first = new;
+    } else {  // all other cases
+        new->next = L->curr;
+        new->prev = L->curr->prev;
+
+        L->curr->prev->next = new;
+        L->curr->prev = new;
+        L->curr = new;
+    }
     // COMPLETE THIS FUNCTION
```
Comment (kotarou3) [inline week03/DLList.c]: You can remove these comments now

File: week03/DLList.c [hunk 5]
```diff
@@ -209,7 +210,24 @@ int DLListMoveTo(DLList L, int i)
...
+    L->nitems++;
```
Comment (kotarou3) [inline week03/DLList.c]: I'd prefer you stick to using the pre- versions rather than the post- versions

**Comment by minight** [comment]
tl;dr fixed
